src/components/HistoryTable/HistoryTableContext.tsx

```diff
@@ -59,34 +63,24 @@ const HistoryTableProvider: React.FC<HistoryTableProviderProps> = ({
       columnLimit,
       historyTableFilters,
       commitCount,
+      rowSizes,
     },
     dispatch,
   ] = useReducer(reducer, {
     ...initialState,
   });
-  const itemHeight = (index: number) => {
-    if (processedCommits[index]) {
-      switch (processedCommits[index].type) {
-        case rowType.COMMIT:
-          return 120;
-        case rowType.DATE_SEPARATOR:
-          return 40;
-        case rowType.FOLDED_COMMITS:
-          return 40;
-        default:
-          return 100;
-      }
-    } else {
-      return 120;
-    }
-  };
```

Comment:
How do you feel about not creating a separate array for `rowSizes` and store it as a field within processedCommits? 
It would reduce some memory allocation since we could calculate the values only when they are needed. It's also one less array to update and iterate through.